src/codegen/globals.rs

```diff
@@ -71,6 +71,12 @@ pub fn insert_globals<'a>(
     }
 }
 
+fn write_globals_len<'a>(llvm_ctx: &'a LLVMCtx, llvm_module: &'a LLVMModule, globals_len: u32) {
```

Comment:
Super duper nit: shouldn't this go after the next function?

Reply:
Happy to reorder however you want. Which functions do you want this between?

Reply:
I'm going to address the unit tests in here before merging. I might have missed something.

Reply:
Just above initializer_to_value?

Reply:
Done! Going to take a look at the unit test stuff for this PR shortly.